lib/ddtrace/contrib/rails/patcher.rb

```diff
@@ -55,6 +57,19 @@ def add_middleware(app)
           end
         end
 
+        def add_logger(app)
+          # check if lograge key exists
+          if app.config.respond_to?(:lograge) && app.config.lograge.enabled
+            Datadog::Contrib::Rails::LogInjection.add_lograge_logger(app)
+          # if lograge isn't set, check if tagged logged is enabed.
+          # if so, add proc that injects trace identifiers for tagged logging.
+          elsif (logger = app.config.logger) && logger.is_a?(::ActiveSupport::TaggedLogging)
+            Datadog::Contrib::Rails::LogInjection.add_as_tagged_logging_logger(app)
+          else
+            Datadog.logger.warn('Unabe to enable Datadog Trace context, Logger is not supported')
```

Comment:
I suggest we log the class type of `logger` in this message.

Reply:
i am a bit hestitent to use the application's logger directly as it's not clear to me that `warn`, `debug`, `info` etc would necessarily be defined. I suppose I can just write some defensiveness into the call tho `logger.warn(...) if logger && logger.respond_to?(:warn)`

Reply:
My bad, I mean in the message:
`'Unable to enable Datadog Trace context, Logger #{logger_class_we_werent_able_to_instrument_here} is not supported'`

Reply:
👍 makes sense, updated